query-service-impl/src/main/java/org/hypertrace/core/query/service/pinot/QueryRequestToPinotSQLConverter.java

```diff
@@ -40,14 +42,15 @@
   private final ViewDefinition viewDefinition;
   private final PinotFunctionConverter functionConverter;
   private final Joiner joiner = Joiner.on(", ").skipNulls();
+  private Map<String,Integer> selectionMap = new HashMap<String,Integer>();
 
   QueryRequestToPinotSQLConverter(
       ViewDefinition viewDefinition, PinotFunctionConverter functionConverter) {
     this.viewDefinition = viewDefinition;
     this.functionConverter = functionConverter;
   }
 
-  Entry<String, Params> toSQL(
+  Pair<Map<String ,Integer>,Entry<String, Params>> toSQL(
```

Comment:
there's the pair! This actually highlights the problem with it too - I have no idea, as a reader, what this is supposed to represent. Pair does not convey semantic information.

Reply:
I need to return two values from the function. Pair would be better ig than list/map/collection.

Reply:
updated the code